pkg/internal/resource_with_tag.go

```diff
@@ -0,0 +1,72 @@
+package internal
```

Comment:
Not sure about this files location. Unfortunately within `pkg/apis/core/v1/internal` the `AutoTag` exec part wouldn't have access to the `ResourceWithTag` struct.